src/Runner.Worker/ActionManager.cs

```diff
@@ -247,7 +247,9 @@ private async Task<PrepareActionsState> PrepareActionsRecursiveAsync(IExecutionC
                                 {
                                     _cachedEmbeddedPreSteps[parentStepId] = new List<Pipelines.ActionStep>();
                                 }
-                                _cachedEmbeddedPreSteps[parentStepId].Add(action);
+                                var clonedAction = action.Clone() as Pipelines.ActionStep;
```

Comment:
Need to clone so we can update the condition without modify other copies of it.

Reply:
can you add a comment in the source code